lib/Dialect/CGGI/Conversions/CGGIToJaxite/CGGIToJaxite.cpp

```diff
@@ -140,6 +142,57 @@ struct ConvertCGGIToJaxiteLut3Op : public OpConversionPattern<cggi::Lut3Op> {
   }
 };
 
+struct ConvertCGGIToJaxitePmapLut3Op
+    : public OpConversionPattern<cggi::PackedLut3Op> {
+  ConvertCGGIToJaxitePmapLut3Op(mlir::MLIRContext *context)
+      : OpConversionPattern<cggi::PackedLut3Op>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      cggi::PackedLut3Op op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    FailureOr<Value> resultServerKey =
+        getContextualJaxiteArg<jaxite::ServerKeySetType>(op.getOperation());
+    if (failed(resultServerKey)) return resultServerKey;
+    Value serverKey = resultServerKey.value();
+
+    FailureOr<Value> resultParams =
+        getContextualJaxiteArg<jaxite::ParamsType>(op.getOperation());
+    if (failed(resultParams)) return resultParams;
+    Value params = resultParams.value();
+    auto A = op.getA();
+    auto B = op.getB();
+    auto C = op.getC();
+    auto truthTableValues = op.getLookupTables();
+    SmallVector<Value> lut3_args;
+    for (int i = 0; i < truthTableValues.size(); ++i) {
+      uint8_t truthTableValue = static_cast<uint8_t>(
+          cast<IntegerAttr>(truthTableValues[i]).getUInt());
+      auto tt = b.create<arith::ConstantOp>(
+          op.getLoc(), b.getIntegerAttr(b.getI8Type(), truthTableValue));
+      auto extractionIndex =
+          b.create<arith::ConstantOp>(op->getLoc(), b.getIndexAttr(i));
+      auto extractOpA = b.create<tensor::ExtractOp>(
+          op->getLoc(), A, extractionIndex.getResult());
+      auto extractOpB = b.create<tensor::ExtractOp>(
+          op->getLoc(), B, extractionIndex.getResult());
+      auto extractOpC = b.create<tensor::ExtractOp>(
+          op->getLoc(), C, extractionIndex.getResult());
+      auto lut3ArgsOp = b.create<jaxite::Lut3ArgsOp>(
+          op.getLoc(), extractOpA.getResult(), extractOpB.getResult(),
+          extractOpC.getResult(), tt.getResult());
```

Comment:
also a small readability thing you don't need the `getResult()` for `extractOpX`s since an operation with one result can be implcitly cast to a Value

Reply:
Done!